src/network/networkd-setlink.c

```
@@ -1341,6 +1346,97 @@ int link_up_or_down_now(Link *link, bool up) {
         return 0;
 }
 
+typedef struct SetLinkVarlinkContext {
+        Link *link;
+        sd_varlink *vlink;
+        bool up;
+} SetLinkVarlinkContext;
+
+static void set_link_varlink_context_destroy(SetLinkVarlinkContext *ctx) {
+        assert(ctx);
+
+        if (ctx->vlink)
+                sd_varlink_unref(ctx->vlink);
+        if (ctx->link)
+                link_unref(ctx->link);
+        free(ctx);
+}
+
+static int link_up_or_down_now_varlink_handler(sd_netlink *rtnl, sd_netlink_message *m, SetLinkVarlinkContext *ctx) {
+        int r;
+
+        assert(m);
+        assert(ctx);
+
+        Link *link = ASSERT_PTR(ctx->link);
+        sd_varlink *vlink = ASSERT_PTR(ctx->vlink);
+        bool up = ctx->up;
+
+        assert(link->set_flags_messages > 0);
+
+        link->set_flags_messages--;
+
+        r = sd_netlink_message_get_errno(m);
+        if (r < 0)
+                log_link_message_warning_errno(link, m, r, "Could not bring %s interface", up_or_down(up));
+
+        if (r >= 0 && !IN_SET(link->state, LINK_STATE_FAILED, LINK_STATE_LINGER)) {
+                int k = link_call_getlink(link, get_link_update_flag_handler);
```

Please also propagate the failure to the varlink method caller.
Hence, please use `r` rather than `int k`.